src/messaging/messaging.ts

```diff
@@ -245,15 +301,26 @@ export class Messaging implements MessagingInterface {
       throw new FirebaseMessagingError(
         MessagingClientErrorCode.INVALID_ARGUMENT, 'dryRun must be a boolean');
     }
-    return this.getUrlPath()
-      .then((urlPath) => {
-        const request: { message: Message; validate_only?: boolean } = { message: copy };
-        if (dryRun) {
-          request.validate_only = true;
-        }
-        return this.messagingRequestHandler.invokeRequestHandler(FCM_SEND_HOST, urlPath, request);
+
+    const clientMessage = this.convertToClientMessage(copy);
+
+    const client = new FcmServiceClient({
+      credentials: this.getAppKey(),
```

Comment:
We should try to set `projectId` here too, and remove `parent` field in the request.

Reply:
The parent field is a required part of the request ([See here](https://github.com/firebase/firebase-admin-node/blob/cc435abbb87df2db98330afc620c3551d94ea7c6/src/generated/messaging/src/v1/fcm_service_client.ts#L293)), so it's unfortunately not possible to set in the constructor.

Reply:
The doc comments say so, but it's really optional in the type declartion:

https://github.com/firebase/firebase-admin-node/blob/cc435abbb87df2db98330afc620c3551d94ea7c6/src/generated/messaging/protos/protos.d.ts#L76-L79

Shouldn't be required to specify it.

Reply:
Even when including the `projectId` in the constructor for the client, omitting the `parent` field (the project ID) in the message request results in the generated client throwing an error, specifically stating that it `Cannot build HTTP request`. I'm assuming that the generated type declaration is incorrect here.